src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestBuilder.java

```diff
@@ -439,24 +442,16 @@ private String getVsTestPath(Node builtOn, TaskListener listener, EnvVars env) {
             testFile = replaceMacro(testFile, env);
 
             if (!StringUtils.isBlank(testFile)) {
-
-                for (String file : expandFileSet(workspace, testFile)) {
-                    args.add(appendQuote(file));
+                try {
+                    for (FilePath filePath : workspace.list(testFile)) {

```

Comment:
Ha, a nested for each loop inside another for each loop which was already inside a while loop 😒